### ISSUE TITLE
`OneCycleLR` documentation misleading about optimizer

### ISSUE THREAD

**Issue by carbocation** [pytorch/pytorch#89358]
### 📚 The doc issue

The [OneCycleLR documentation](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.OneCycleLR.html) provides example code that doesn't trigger `optimizer.step()`:

```py
data_loader = torch.utils.data.DataLoader(...)
optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
scheduler = torch.optim.lr_scheduler.OneCycleLR(optimizer, max_lr=0.01, steps_per_epoch=len(data_loader), epochs=10)
for epoch in range(10):
    for batch in data_loader:
        train_batch(...)
        scheduler.step()
```

This is confusing, because the `OneCycleLR` scheduler does engulf the optimizer and could therefore plausibly trigger the `optimizer.step()` call for the user. So the omission seems intentional at first glance. But calling `optimizer.step()` [is in fact necessary](https://stackoverflow.com/q/59996859/199475).

### Suggest a potential alternative/fix

The `OneCycleLR` documentation could just state that calling `optimizer.step()` is still needed, or could show the `optimizer.step()` call in the pseudocode.

cc @svekars @carljparker @vincentqb @jbschlosser @albanD

**Comment by albanD** [comment]
We should update the code sample to call the original optimizer indeed. Could you send a PR doing that?
